src/WebJobs.Script/Workers/Rpc/Configuration/DefaultWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    // This class resolves worker configurations by scanning the "workers" directory within the Host for worker config files.
+    internal class DefaultWorkerConfigurationResolver : IWorkerConfigurationResolver
+    {
+        private readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
+
+        public DefaultWorkerConfigurationResolver(IConfiguration configuration, ILogger logger)
```

Comment:
I missed some of this in the first pass -- but we're trying to move away from passing in IConfiguration or IEnvironment to any classes. They aren't very descriptive -- like right here you have no idea as a consumer what part of IConfiguration is needed.

Instead, using some kind of IOptions<> object that builds the options (using IConfiguration) before passing them in is a much nicer contract and much easier to test/read.

Highly recommend moving away from this where you can. I'll call out some others as I see them.

Reply:
If it works, I would prefer to address this issue in a separate PR. I have created an issue - https://github.com/Azure/azure-functions-host/issues/11185 for it and added it to the "Decoupling workers" milestone. This PR already contains a lot of logic and tests and it is also against a feature branch, so I am not merging changes into dev.
	
However, if you are really concerned about it, I can raise another PR with the fix targeting this PR branch (`surgupta/decoupling-workers`). We can merge that PR first before completing this one. Though it is not my preferred approach, as this PR is becoming long-standing and we are already using this pattern in multiple places in Host code, but I can do it if necessary.

Reply:
Discussed offline. Will be implementing the improvement first.